Add XLSX support and no-header option to lead upload

AgbediaSamuel opened the pull request:
Added support for uploading and parsing .xlsx files in addition to CSVs in the lead file upload and queue pages. Introduced a 'File has no column headers' option, allowing users to map columns when the file lacks headers. Updated UI labels and mapping logic to accommodate both file types and header scenarios.

AgbediaSamuel committed:
Add XLSX support and no-header option to lead upload

Added support for uploading and parsing .xlsx files in addition to CSVs in the lead file upload and queue pages. Introduced a 'File has no column headers' option, allowing users to map columns when the file lacks headers. Updated UI labels and mapping logic to accommodate both file types and header scenarios.

AgbediaSamuel committed:
Add XLSX file support to lead and vendor forms

Enhanced LeadFormPage and externalVendor forms to accept and preview both CSV and XLSX files. Updated file validation, preview, and UI labels to reflect support for Excel files, improving flexibility for users uploading lead data.

IamWafula commented on src/pages/supervisorPages/uploadLeadFile.tsx:
```diff
@@ -808,16 +901,45 @@ const UploadLeadFile = () => {
                     }}
                   >
-                    <Input id="csv-file" type="file" accept=".csv" onChange={handleFileUpload} className="pt-0 hidden" />
+                    <Input id="csv-file" type="file" accept=".csv,.xlsx" onChange={handleFileUpload} className="pt-0 hidden" />
```
We should add support for `.xls` also.

AgbediaSamuel committed:
Improve Excel file detection in upload page

used regex to match .xls file types

AgbediaSamuel committed:
Update uploadLeadFile.tsx

Updated this file to include changes that match what's in production - new homebound fields and base fields replacing the monolith predefined fields

AgbediaSamuel committed:
Merge branch 'production' into xlsx-+-no-column-headers-support

AgbediaSamuel committed:
Update uploadLeadFileQueue.tsx

Made the changes to upload file consistent with the queue version for easy switch

AgbediaSamuel committed:
Add has_no_headers to upload FormData

Include the has_no_headers flag (as JSON.stringify(hasNoHeaders)) in FormData for various upload flows across UploadLeadFile and UploadLeadFileQueue (initial validation, duplicate checks, final upload) so the backend can handle files without headers. Also remove a duplicated HOMEBOUND_EXTRA_COLUMNS definition and apply minor whitespace/formatting cleanups.

AgbediaSamuel committed:
Minor fixes

Changed the text in the app in some places so it's more generic, refactored  two code blocks into a reusable function because why not? and removed the unnecessary regex matching for if something is xlsx and updated html accept attributes for what kind of files can be uploaded on specific pages.

AgbediaSamuel committed:
Reverted the url for the bakend

so platform doesn't break when its deployed to production
